src/common/subgraphMapper.ts

```diff
@@ -137,48 +85,65 @@ export const mapMarketsArrayToInterface = (response: any): Market[] | undefined
-  if (response === null) return undefined;
+export const mapSingleOrderToInterface = (
+  orderResponse: any,
+  depositCollateral?: DepositCollateral[] | undefined,
+): Order | undefined => {
+  if (orderResponse === null) return undefined;
   try {
     return {
-      id: response.id,
-      market: response.market ? mapSingleMarketToInterface(response.market) : undefined,
-      user: response.user ? mapSubgraphResponseToAccountInterface(response.user) : undefined,
-      orderType: response.orderType,
-      isLong: response.isLong,
-      isLimitOrder: response.isLimitOrder,
-      triggerAbove: response.triggerAbove,
-      deadline: response.deadline,
-      deadlineISO: response.deadlineISO,
-      deltaCollateral: response.deltaCollateral,
-      deltaSize: response.deltaSize,
-      deltaSizeUsd: response.deltaSizeUsd,
-      expectedPrice: response.expectedPrice,
-      maxSlippage: response.maxSlippage,
-      partnerAddress: response.partnerAddress,
-      executionFee: response.executionFee,
-      txHash: response.txHash,
-      createdTimestamp: response.createdTimestamp,
-      status: response.status,
-      settledTxHash: response.settledTxHash,
-      settledTimestamp: response.settledTimestamp,
-      settledTimestampISO: response.settledTimestampISO,
-      executionPrice: response.executionPrice,
-      settledBy: response.settledBy ? mapSubgraphResponseToAccountInterface(response.settledBy) : undefined,
-      cancellationTxHash: response.cancellationTxHash,
-      positionId: response.position ? response.position.id : undefined,
+      id: orderResponse.id,
+      market: mapSingleMarketToInterface(orderResponse.market),
+      user: mapSubgraphResponseToWalletInterface(orderResponse.user),
+      isLimitOrder: orderResponse.isLimitOrder,
+      deadline: orderResponse.expirationTime,
+      deltaCollateral: orderResponse.deltaCollateral,
+      deltaSize: orderResponse.deltaSize,
+      deltaSizeUsd: orderResponse.deltaSizeUsd,
+      executionFee: orderResponse.collectedFees,
+      txHash: orderResponse.txHash,
+      createdTimestamp: orderResponse.createdTimestamp,
+      status: orderResponse.status,
+      settledTxHash: orderResponse.settledTxHash,
+      settledTimestamp: orderResponse.settledTimestamp,
+      settledTimestampISO: orderResponse.settledTimestampISO,
+      executionPrice: orderResponse.executionPrice,
+      formattedExecutionPrice: formatEther(orderResponse.executionPrice),
+      expectedPrice: orderResponse.acceptablePrice,
+      formateedExpectedPrice: formatEther(orderResponse.acceptablePrice),
+      settledBy: orderResponse.settledBy ? mapSubgraphResponseToWalletInterface(orderResponse.settledBy) : undefined,
+      positionId: orderResponse.position ? orderResponse.position.id : undefined,
+      formattedDeltaSize: formatEther(orderResponse.deltaSize),
+      depositCollateral: depositCollateral,
+      snxAccount: orderResponse.snxAccount.id,
```

Comment:
This should be mapped to the snxAccountId and not the subgraph entity id

Reply:
Done